ui/raid/import_export.ts

```diff
@@ -142,301 +136,799 @@ class RaidWCLImporter extends Importer {
+	
+	constructor(data: wclPlayer, simUI: RaidSimUI, faction: Faction = Faction.Unknown) {
+		this.simUI = simUI;
+		
+		this.name = data.name;
+		this.gear = data.gear;
+		this.icon = data.icon;
+		this.id = data.id;
+		this.type = data.type;
+		this.talents = data.talents;
+		this.wclSpec = data.icon.split("-")[1];
+		this.faction = faction;
+		
+		// Prot Paladin's occasionally have a specType of "Protection" instead of "Justicar"?
+		if (this.type === "Paladin" && this.wclSpec === "Protection") {
+			this.wclSpec = "Justicar";
+		}
+		
+		this.spec = specNames[this.wclSpec];
+		this.specType = this.wclSpec + this.type;
+		
+		this.isBuffBot = this.spec === undefined;
+		
+		this.isPartyBuffer = this.getPartyAuraIds().length > 1;
+		
+		this.sortPriority = specSortPriority[this.wclSpec] ?? 99;
+	}
+	
+	public getPlayer(): Player | undefined {
+		
+		if (this.isBuffBot) {
+			return;
+		}
+		
+		let player = Player.create();
+		
+		const specFuncs = specTypeFunctions[this.spec];
+		
+		const matchingPreset = this.getMatchingPreset();
+		
+		if (matchingPreset === undefined) {
+			console.error("Could not find matching preset for non buff bot", {
+				"name": this.name,
+				"spec": this.spec,
+				"type": this.type,
+				"talents": this.talents,
+			});
+			return;
+		}
+		
+		player = withSpecProto(this.spec, player, matchingPreset.rotation, specFuncs.talentsCreate(), matchingPreset.specOptions);
+		
+		player.talentsString = matchingPreset.talents;
+		player.consumes = matchingPreset.consumes;
+		
+		player.name = this.name;
+		player.class = nameToClass(this.type);
+		player.equipment = this.getEquipment();
+		player.race = matchingPreset.defaultFactionRaces[this.faction];
+		
+		return player;
+	}
+	
+	public getBuffBot(): BuffBot | undefined {
+		
+		if (!this.isBuffBot) {
+			return;
+		}
+		
+		const botID = buffBotNames[this.specType];
+		
+		if (botID == null) {
+			console.error("Buff Bot Spec not implemented: ", this.specType);
+			return;
+		}
+		
+		const bot = BuffBot.create();
+		bot.id = botID;
+		bot.raidIndex = -1; // Default it for now. // numPlayers
+		
+		return bot;
+	}
+	
+	public getPartyAuraIds() {
+		
+		const allSpecClassAuras: any = {
+			"Paladin": [
+				19746, // Concentration Aura
+				27149, // Devotion Aura,
+				27150, // Retribution Aura
+			],
+			"Warrior": [
+				2048,  // Battle Shout
+				469, // Commanding Shout
+			],
+			"Warlock": [
+				27268, // Pet Imp: Blood Pact
+				18696, // Improved Imp: Blood Pact
+			],
+		};
+		
+		// Reused for the plethora of Feral specs.
+		const feralDruidSpecAuras = [
+			24932, // Improved Leader of the Pack // at least 0,32,0
+			// 17007, // Leader of the Pack // at least 0,31,0
+		];
+		
+		// TODO: Could additionally filter out buff IDs based on minimum req talent strings?
+		const specSpecificAuras: any = {
+			"RetributionPaladin": [
+				20092, // Improved Retribution Aura // at least 0,0,16
+				20218, // Sanctity Aura // at least 0,0,21
```

Comment:
some prot paladin spec into sanc/imp sanc

Reply:
Interesting. I hadn't seen any of the common talents with it, so I had just assumed it was Ret only. I think the talent minimums is a good Avenue to expand this into in the future though.